vyper/ast.py

```diff
@@ -0,0 +1,278 @@
+from itertools import (
+    chain,
+)
+from typing import (
+    Any,
+    List as ListTyping,
+)
+
+from vyper.exceptions import (
+    CompilerPanic,
+)
+
+
+class VyperNode:
+    __slots__ = ('node_id', 'source_code', 'col_offset', 'lineno')
+    ignored_fields = ['ctx', ]
```

Comment:
Classes should not use mutable data structures as class properties.

[flake8-bugbear](https://github.com/PyCQA/flake8-bugbear) will catch these types of mistakes in CI.

Reply:
Well that's odd flake8-bugbear is enabled, didn't catch it.

Reply:
I was wrong.  It only applies this check to function argument defaults.